gpt_index/indices/query/retriever_query_engine.py

```diff
@@ -0,0 +1,121 @@
+from typing import Any, Dict, List, Optional, Sequence
+from gpt_index.data_structs.node_v2 import NodeWithScore
+from gpt_index.indices.common.base_retriever import BaseRetriever
+from gpt_index.indices.postprocessor.node import BaseNodePostprocessor
+from gpt_index.indices.query.base import BaseQueryEngine
+from gpt_index.indices.query.schema import QueryBundle
+from gpt_index.indices.response.response_synthesis import ResponseSynthesizer
+from gpt_index.indices.response.type import ResponseMode
+from gpt_index.indices.service_context import ServiceContext
+from gpt_index.optimization.optimizer import BaseTokenUsageOptimizer
+from gpt_index.prompts.prompts import (
+    QuestionAnswerPrompt,
+    RefinePrompt,
+    SimpleInputPrompt,
+)
+from gpt_index.response.schema import RESPONSE_TYPE
+
+
+class RetrieverQueryEngine(BaseQueryEngine):
```

Comment:
are we exposing this to the user? i'm assuming the answer is yes? 

i think we should expose the retriever classes (if the user just wants to do retrieval), and ResponseSynthesizer, and also this query engine class as user-facing.

Reply:
Yes